### PR TITLE
Fix kotlin test CI and android TV cast example CI

### PR DESCRIPTION
**Pull request by yunhanw-google**
-- previous https://github.com/project-chip/connectedhomeip/pull/25636 does not update python wrapper for kotlin im test runtest function, need add missing piece so that CI can pass
-- previous https://github.com/project-chip/connectedhomeip/pull/25505 does not update the subscriptionId in OnEstablishedCallback in example android tv cast side. need update it.

### CODE REVIEW

**Review by joonhaengHeo — APPROVED** [review]

File: scripts/tests/run_java_test.py [hunk 1]
```diff
@@ -107,6 +108,15 @@ def main(app: str, app_args: str, tool_path: str, tool_cluster: str, tool_args:
         except Exception as e:
             logging.error(e)
             sys.exit(1)
+    elif tool_cluster == 'im':
```
Comment (yufengwangca) [inline scripts/tests/run_java_test.py:111]: It looks all those four commands 'onnetwork-long-im-invoke/onnetwork-long-im-read/onnetwork-long-im-write/onnetwork-long-im-subscribe' are implemented within pairing cluster in the code

File: .github/workflows/tests.yaml [hunk 2]
```diff
@@ -609,7 +609,7 @@ jobs:
                      --app out/linux-x64-all-clusters-ipv6only-no-ble-no-wifi-tsan-clang-test/chip-all-clusters-app \
                      --app-args "--discriminator 3840 --interface-id -1" \
                      --tool-path out/linux-x64-java-matter-controller \
-                     --tool-cluster "pairing" \
+                     --tool-cluster "im" \
```
Comment (yufengwangca) [inline .github/workflows/tests.yaml:612]: I agree those IM commands (read/write/subscribe/invoke) should not under 'pairing' cluster, but it looks they are still implemented as the commands within the 'pairing' cluster

File: scripts/tests/java/commissioning_test.py [hunk 3]
```diff
@@ -93,6 +93,26 @@ def RunTest(self):
...
+        elif self.command_name == 'onnetwork-long-im-read':
+            logging.info("Testing pairing onnetwork-long-im-read")
+            code = self.TestCmdOnnetworkLongImRead(self.nodeid, self.setup_pin_code, self.discriminator, self.timeout)
+            if code != 0:
+                raise Exception(f"Testing pairing onnetwork-long-im-read failed with error {code}")
+        elif self.command_name == 'onnetwork-long-im-subscribe':
+            logging.info("Testing pairing onnetwork-long-im-subscribe")
+            code = self.TestCmdOnnetworkLongImSubscribe(self.nodeid, self.setup_pin_code, self.discriminator, self.timeout)
+            if code != 0:
+                raise Exception(f"Testing pairing onnetwork-long-im-subscribe failed with error {code}")
```
Comment (yufengwangca) [inline scripts/tests/java/commissioning_test.py:115]: Why those commands are triggers from im cluster in the test.yaml, but implemented within commissioning_test?

`--tool-cluster "im" \`

It looks you have already added 'scripts/tests/java/im_test.py' for those tests.

**Comment by yufengwangca** [comment]
I saw all of IM commands (read/write/invoke/subscribe) are implemented within pairing cluster, but tested under another cluster 'im', under which clusters those commands are implemented in chip-tool?  we need to align with chip--tool. Finally, we should auto-generated the command handlers from its xml cluster definition file

File: scripts/tests/java/commissioning_test.py (hunk 3 above)
Comment (yunhanw-google) [reply scripts/tests/java/commissioning_test.py:115]: need to be removed. thanks

File: .github/workflows/tests.yaml (hunk 2 above)
Comment (yunhanw-google) [reply .github/workflows/tests.yaml:612]: removed. thanks

File: scripts/tests/java/commissioning_test.py (hunk 3 above)
Comment (yunhanw-google) [reply scripts/tests/java/commissioning_test.py]: remove, thanks

File: scripts/tests/run_java_test.py (hunk 1 above)
Comment (yunhanw-google) [reply scripts/tests/run_java_test.py:111]: update,d thanks

File: scripts/tests/java/im_test.py [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+        elif self.command_name == 'onnetwork-long-im-read':
+            logging.info("Testing pairing onnetwork-long-im-read")
+            code = self.TestCmdOnnetworkLongImRead(self.nodeid, self.setup_pin_code, self.discriminator, self.timeout)
+            if code != 0:
+                raise Exception(f"Testing pairing onnetwork-long-im-read failed with error {code}")
+        elif self.command_name == 'onnetwork-long-im-subscribe':
+            logging.info("Testing pairing onnetwork-long-im-subscribe")
+            code = self.TestCmdOnnetworkLongImSubscribe(self.nodeid, self.setup_pin_code, self.discriminator, self.timeout)
+            if code != 0:
+                raise Exception(f"Testing pairing onnetwork-long-im-subscribe failed with error {code}")
```
Comment (yufengwangca) [inline scripts/tests/java/im_test.py:118]: I am still confused with this implementation. It looks the IM cluster and pairing cluster are mangled together in test "Testing pairing onnetwork-long-im-subscribe"

From the implementation side, I saw all IM commands are registered to the 'im' cluster

`  commandManager.register("im", getImCommands(controller, credentialsIssuer))`

But implemented as a 'pairing' command

```
class PairOnNetworkLongImInvokeCommand(
  controller: ChipDeviceController, credsIssue: CredentialsIssuer?
) : PairingCommand
```

Comment (yunhanw-google) [reply scripts/tests/java/im_test.py:118]: yes, in current test, we always starts from fresh state, so we need pair first, then run im commands

**Comment by yunhanw-google** [comment]
> I saw all of IM commands (read/write/invoke/subscribe) are implemented within pairing cluster, but tested under another cluster 'im', under which clusters those commands are implemented in chip-tool? we need to align with chip--tool. Finally, we should auto-generated the command handlers from its xml cluster definition file

these should be tested in im category, however, we don't have enough bandwidth to figure out how to extract the certs info and setup case connection in these im tests, in addition, standalone case api in jni is not ready. Would file the ticket on it.

We may use zap to generate a list of test commands later, file the ticket as well. 

thanks

**Comment by yufengwangca** [comment]
> > I saw all of IM commands (read/write/invoke/subscribe) are implemented within pairing cluster, but tested under another cluster 'im', under which clusters those commands are implemented in chip-tool? we need to align with chip--tool. Finally, we should auto-generated the command handlers from its xml cluster definition file
> 
> these should be tested in im category, however, we don't have enough bandwidth to figure out how to extract the certs info and setup case connection in these im tests, in addition, standalone case api in jni is not ready. Would file the ticket on it.
> 
> #25649
> 
> We may use zap to generate a list of test commands later, file the ticket as well. #25650
> 
> thanks

https://github.com/project-chip/connectedhomeip/issues/25650 is a FR and not imminent,  I updated the description of 

https://github.com/project-chip/connectedhomeip/issues/25649, this is not about test, the problem is in IM cluster commands implementation, currently, all IM cluster commands are registered under cluster 'im', but implemented as 'pairing' cluster command. The behavior is undefined, if we register a command in one cluster, but implemented it under another cluster. Lets follow-up this after this PR lands

**Review by yufengwangca — APPROVED** [review]

**Pull request merged**